store.go

```diff
@@ -3054,13 +3054,15 @@ func (s *store) Layers() ([]Layer, error) {
 	if err != nil {
 		return nil, err
 	}
-	if err := lstore.LoadLocked(); err != nil {
-		return nil, err
-	}
-	layers, err := lstore.Layers()
-	if err != nil {
-		return nil, err
-	}
+
+	layers, err := func() ([]Layer, error) {
+		lstore.RLock()
+		defer lstore.Unlock()
+		if err := lstore.Load(); err != nil {
+			return nil, err
+		}
+		return lstore.Layers()
+	}()
```

Comment:
Looks like the check for `err` got lost here, otherwise LGTM.

Reply:
Done. Also had to use a writer lock because of Load.

Reply:
That was the only caller for LoadLocked().  We may want to try removing that at some point.

Reply:
want me to drop a `TODO`?

Reply:
Not a blocker, but if you need to revise this for any reason, that's probably a good idea.  Thanks!